File: src/problem022/solution022.py

def alphabetic_value(text):
    return sum([ord(char) - ord('A') + 1 for char in text.upper()])

def main():
    names = sorted([name.strip('"') for name in open('names.txt').read().split(',')])
    print(sum([(idx + 1) * alphabetic_value(name) for idx, name in enumerate(names)]))

if __name__ == '__main__':
    main()

File: src/problem022/solution022.hs
import Data.Char (ord)
import Data.List (dropWhileEnd, sort, zipWith)

main = do
    content <- readFile "names.txt"
    print . sum . zipWith score [1..] . sort .
        map (strip '"') . split ',' $ content

score idx = (*idx) . (sum . map ((+(1-ord 'A')) . ord))

split chr "" = []
split chr str = part : (split chr $ dropWhile (==chr) rest)
    where (part, rest) = break (==chr) str

strip chr = dropWhileEnd (==chr) . dropWhile (==chr)
